packages/tools/devtools/devtools-view/src/components/ContainerRuntimeView.tsx

```diff
@@ -0,0 +1,94 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Tree as FluentTree } from "@fluentui/react-components";
+import {
+	GetRootDataVisualizations,
+	type HasContainerKey,
+	type ISourcedDevtoolsMessage,
+	type InboundHandlers,
+	RootDataVisualizations,
+	type RootHandleNode,
+	handleIncomingMessage,
+} from "@fluidframework/devtools-core/internal";
+import React from "react";
+
+import { useMessageRelay } from "../MessageRelayContext.js";
+
+import { Waiting } from "./Waiting.js";
+import { TreeDataView } from "./data-visualization/index.js";
+
+const loggingContext = "INLINE(VIEW)";
+
+/**
+ * {@link ContainerRuntimeView} input props.
+ */
+export type ContainerRuntimeViewProps = HasContainerKey;
+
+/**
+ * Displays the data inside a container.
+ *
+ * @remarks
+ *
+ * Dispatches container runtime rendering based on those provided view {@link TreeDataView}.
+ */
+export function ContainerRuntimeView(props: ContainerRuntimeViewProps): React.ReactElement {
+	const { containerKey } = props;
+
+	const messageRelay = useMessageRelay();
+
+	const [rootDataHandles, setRootDataHandles] = React.useState<
+		Record<string, RootHandleNode> | undefined
+	>();
+
+	React.useEffect(() => {
+		const inboundMessageHandlers: InboundHandlers = {
+			[RootDataVisualizations.MessageType]: async (untypedMessage) => {
+				const message = untypedMessage as RootDataVisualizations.Message;
+
+				if (message.data.containerKey === containerKey) {
+					setRootDataHandles(message.data.visualizations);
+
+					return true;
+				} else {
+					return false;
+				}
+			},
+		};
+
+		function messageHandler(message: Partial<ISourcedDevtoolsMessage>): void {
+			handleIncomingMessage(message, inboundMessageHandlers, {
+				context: loggingContext,
+			});
+		}
+
+		messageRelay.on("message", messageHandler);
+
+		// POST Request for DDS data in container.
+		messageRelay.postMessage(
```

Comment:
Not for this PR, but we should probably add a timeout for this request in this component. If we fail to get a response from the client, then we will just show a spinner forever. In some other places, we've added a timeout and display an error message if we don't get a response within the specified time threshold (with a "try again" button to re-attempt the request for data).